dnsutil/zone.go

```diff
@@ -0,0 +1,209 @@
+package dnsutil
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+type zoneGenData struct {
+	Origin     string
+	DefaultTtl uint32
+	Records    []dns.RR
+}
+
+func (z *zoneGenData) Len() int      { return len(z.Records) }
+func (z *zoneGenData) Swap(i, j int) { z.Records[i], z.Records[j] = z.Records[j], z.Records[i] }
+func (z *zoneGenData) Less(i, j int) bool {
+	a, b := z.Records[i], z.Records[j]
+	fullA, fullB := AddDomain(z.Origin, a.Header().Name), AddDomain(z.Origin, b.Header().Name)
+	if fullA != fullB {
+		return zoneLabelLess(fullA, fullB)
+	}
+	rrtypeA, rrtypeB := a.Header().Rrtype, b.Header().Rrtype
+	if rrtypeA != rrtypeB {
+		return zoneRrtypeLess(rrtypeA, rrtypeB)
+	}
+	if rrtypeA == dns.TypeA {
+		ta2, tb2 := a.(*dns.A), b.(*dns.A)
+		ipa, ipb := ta2.A.To4(), tb2.A.To4()
+		if ipa == nil || ipb == nil {
+			log.Fatalf("should not happen: IPs are not 4 bytes: %#v %#v", ta2, tb2)
+		}
+		return bytes.Compare(ipa, ipb) == -1
+	}
+	if rrtypeA == dns.TypeMX {
+		ta2, tb2 := a.(*dns.MX), b.(*dns.MX)
+		pa, pb := ta2.Preference, tb2.Preference
+		return pa < pb
+	}
+	return a.String() < b.String()
+}
+
+func WriteZoneFile(w io.Writer, records []dns.RR, origin string, defaultTtl uint32) error {
```

Comment:
If we are going to add something like this, we should focus on making something that allows users to override the way a RR is printed. I.e. stuff like: omit class or ttl, only print the rdata (or a subset of the rdata). I.e. something super flexible. I have no good ideas (yet) on how something like that should look.